pypesto/sampling/pymc3.py

```diff
@@ -167,25 +167,23 @@ class TheanoLogProbabilityGradient(tt.Op):
     Theano wrapper around a (non-normalized) log-probability gradient function.
     This Op will be called with a vector of values and also return a vector of
     values - the gradients in each dimension.
-
-    Parameters
-    ----------
-    problem:
-        The `pypesto.Problem` to analyze.
-    beta:
-        Inverse temperature (e.g. in parallel tempering).
     """
 
     itypes = [tt.dvector]  # expects a vector of parameter values when called
     otypes = [tt.dvector]  # outputs a vector (the log prob grad)
 
-    def __init__(self, problem: Problem, beta: float = 1.):
-        self._objective: Objective = problem.objective
-        self._log_prob_grad = \
-            lambda x: - beta * self._objective(x, sensi_orders=(1,))
+    def __init__(self, logp_op: TheanoLogProbability):
+        self._logp_op: TheanoLogProbability = logp_op
+        self._objective = logp_op._objective
+        self._beta = logp_op._beta
 
     def perform(self, node, inputs, outputs, params=None):
         theta, = inputs
-        # calculate gradients
-        log_prob_grad = self._log_prob_grad(theta)
-        outputs[0][0] = log_prob_grad
+
+        cached_theta = self._logp_op._cached_theta
```

Comment:
sorry, in the meantime we had to change the file structure a bit, these two classes are now (slightly changed) in `theano.py`.